p2p/module.go

```diff
@@ -334,13 +351,56 @@ func (m *p2pModule) handleAppData(data []byte) error {
 		return fmt.Errorf("decoding network message: %w", err)
 	}
 
+	if ok, err := m.checkNonce(networkMessage.GetNonce()); err != nil {
+		return fmt.Errorf("handling network data: %w", err)
+	} else if !ok {
+		// skip passing redundant message to application layer
+		return nil
+	}
+
+	// DISCUSS_THIS_COMMIT: is there an explicit reason for constructing a new
```

Comment:
I do not remember the exact reasoning. Your example/reasoning seems accurate, and I believe our attempt at making sure the nonce wasn't propagated was once of the reasons.

I believe one of the reasons was to pass a structure by value, rather than by reference, in case downstream functionality may try to modify it.